flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultSessionJobValidator.java

```diff
@@ -17,12 +17,18 @@
 
 package org.apache.flink.kubernetes.operator.validation;
 
+import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
 import org.apache.flink.kubernetes.operator.crd.FlinkSessionJob;
 
 import java.util.Optional;
 
 /** Default validator implementation for {@link FlinkSessionJob}. */
-public class DefaultSessionJobValidator implements FlinkResourceValidator<FlinkSessionJob> {
+public class DefaultSessionJobValidator implements FlinkResourceValidator {
```

Comment:
Now that we have both methods in the interface can we drop the `DefaultSessionJobValidator` and just move the logic to the `DefaultDeploymentValidator`?

Reply:
I have dropped the `DefaultSessionJobValidator`. Please see my last commit. The webhook have not changed accordingly. I want to create another PR for it. 
Not I use `validate(FlinkSessionJob sessionJob, Optional<FlinkDeployment> session)` to validate the session job. It will check the relations of both. But for webhook, it may not have access to the secondary resource, Maybe we could only verify the target session job in the webhook and do the full validation in the operator. what do you think of this ?